plugin/kubernetes/external.go

```diff
@@ -67,22 +71,51 @@ func (k *Kubernetes) External(state request.Request) ([]msg.Service, int) {
 			continue
 		}
 
-		for _, ip := range svc.ExternalIPs {
-			for _, p := range svc.Ports {
-				if !(match(port, p.Name) && match(protocol, string(p.Protocol))) {
-					continue
-				}
-				rcode = dns.RcodeSuccess
-				s := msg.Service{Host: ip, Port: int(p.Port), TTL: k.ttl}
-				s.Key = strings.Join([]string{zonePath, svc.Namespace, svc.Name}, "/")
+		services = append(services, svcToMsg(svc, zonePath, port, protocol, k.ttl)...)
+	}
+	if len(services) > 0 {
+		rcode = dns.RcodeSuccess
+	}
+	return services, rcode
+}
 
-				services = append(services, s)
-			}
+func (k *Kubernetes) externalAll(state request.Request) ([]msg.Service, int) {
+
+	serviceList := k.APIConn.ServiceList()
+
+	services := []msg.Service{}
+	zonePath := msg.Path(state.Zone, coredns)
+	rcode := dns.RcodeNameError
```

Comment:
this means, when we don't find any services we return nxdomain for the zone it self? That's not correct, should be nodata (at least)

Reply:
hum. Anyway this `externallAll` should just return the services.